### ISSUE TITLE
Problem with date in different timezone (UTC Offset)

### ISSUE THREAD

**Issue by wellgrisa** [commenthol/date-holidays-parser#13]
Hello,

We have a timezone system based in different customers within the same codebase. Each customer has its timezone set against itself and we use it together with moment-timezone to deal with their correct and exact time and days (start of, end of and exactly time for each time).

The issue happens when the holiday and the timezone offsets the zulu time per one day, e.g.:
 - checking if `01-01-2018` is a holiday
 - in australia timezone this day is `2017-12-31T13:00:00.000Z`
 - when using hd.isHoliday it uses `const year = toYear(date);` which returns 2017 
 - thus, it gets the holidays set from 2017 which ends up in not being a holiday, once there is no data where the given date is => startDate or < endDate (according from the excerpt of the code)
```
if (hd[j] && date >= hd[j].start && date < hd[j].end)
```

In other words, given:
```
const hd = new Holidays('AU');
const holiday = hd.isHoliday(new Date("2017-12-31T13:00:00.000Z"));
```
holiday should be
```
{ 
  date: '2018-01-01 00:00:00',
  start: 2017-12-31T13:00:00.000Z,
  end: 2018-01-01T13:00:00.000Z,
  name: "New Year's Day",
  type: 'public' 
}
```
However it's returning that this is not a holiday.

**Comment by commenthol** [comment]
Problem is verified and fix is on its way.
Thanks for rising this issue.

**Comment by commenthol** [comment]
@wellgrisa Version 1.4.2 is published which should fix the issue.
